### ISSUE TITLE
Style/For should be marked as unsafe

### ISSUE THREAD

**Issue by diesl** [rubocop/rubocop#10427]
Cop Style/For should be marked as unsafe, because it can break code

Reason might be that scopes are different, as quoted from [Ruby Style Guide](https://rubystyle.guide/#no-for-loops)
> `for` doesn’t introduce a new scope (unlike each) and variables defined in its block will be visible outside it.

--------

## Expected behavior

I expect safe autocorrection not to break my existing ruby code.

## Actual behavior

Running a cop marked as safe breaks my existing ruby code.

## Steps to reproduce the problem

```ruby
for n in 1..3 do
  x = "numbers:" if n == 1
  x = x + " " + n.to_s
  puts x
end

# output is
# numbers: 1
# numbers: 1 2
# numbers: 1 2 3
```

When I run rubocop -a --only Style/IdenticalConditionalBranches, it autocorrects my file to:

```ruby
(1..3).each do |n|
   x = "numbers:" if n == 1
   x = x + " " + n.to_s
  puts x
end

# output is
# numbers: 1
# NoMethodError (undefined method `+' for nil:NilClass)
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.25.1 (using Parser 3.1.0.0, rubocop-ast 1.15.2, running on ruby 2.7.5 x86_64-linux)
  - rubocop-performance 1.13.2
  - rubocop-rails 2.13.2
  - rubocop-rspec 2.8.0
```

**Comment by jonas054** [comment]
I think you mean ` rubocop -a --only Style/For`, but you're right about the reason for the failure. It's the change in scope for the `x` variable in this case.

I agree. Marking it as unsafe is a simple solution.

**Comment by diesl** [comment]
> I think you mean ` rubocop -a --only Style/For`

Sure, I just forgot to change this after c&p a similar issue.

Thanks for fast response and fix 👍
